### ISSUE TITLE
Consider creating an "effective sanitizer configuration" during construction

### ISSUE THREAD

**Issue by mozfreddyb** [WICG/sanitizer-api#153]
I noticed that we do a lot of if/else dance during sanitization that make the algorithms a bit less clear and are sometimes brittle.

I suggest we separate the concern of "taking a config" form the concern of sanitization by building an config dictionary based on the supplied arguments (and filling the missing properties with defaults).

Then we only need to look into that one specific config during sanitization, instead of doing the "is it present, if not look elsewhere" dance.

I believe we discussed this before, but it occurred to me more closely when re-reviewing the spec.

**Comment by otherdaniel** [comment]
For a while, the spec was written that way, and for the reasons you give. I dropped it, since it didn't seem to simplify anything after all: I needed all the same conditions and checks, I just had them in a different place. That way, the "effective configuration" was merely an indirection rather than a valid abstraction. So all it accomplished was that when figuring out what to do, I had to additionally look in another place.

Would be happy to try again, if we have a good take on how to define an effective configuration.
